scripts/apps/dashboard/user-activity/components/UserActivityWidget.tsx

```diff
@@ -224,21 +225,27 @@ export default class UserActivityWidget extends React.Component<{}, IState> {
             groupsData: null,
             loading: false,
         };
+
+        this.refreshItems = this.refreshItems.bind(this);
     }
 
     componentDidMount() {
-        [
+        this.unbindListeners = [
             'item:lock',
             'item:unlock',
             'item:spike',
             'item:move',
             'item:publish',
             'item:update',
         ].map((event) =>
-            this.services.$rootScope.$on(event, this.refreshItems.bind(this)),
+            this.services.$rootScope.$on(event, this.refreshItems),
         );
     }
 
+    componentWillUnmount() {
+        this.unbindListeners.map((unbind) => unbind());
```

Comment:
I know but is there any drawback for using map? I like to stick to just one  method

Reply:
There is, you are making your intention in the code less clear. It's like checking if the string is empty by using `!!myVar` and not indicating that a variable is a string.

When I find such code(someone using map instead of forEach) I start wondering whether the return value was used before and someone accidentally removed it.

Sticking to one method is the silliest reason I've heard :smile:

Reply:
Well in programming using the result of a expression or not is independent of what the function does, and for me they both do the same. Sometimes I do want the value of the expression and sometimes I don't. And `forEach` only gives you the functionality and not the value, that's what I meant by sticking to one method, because it does both. Anyway I can change it to forEach cause I don't care that much to argue about it

Reply:
> Well in programming using the result of a expression or not is independent of what the function does

True, in terms of the output using `map` will get the same result, but it's equally as important that humans understand the code, not only machines. For example, I was wondering whether by using `map` you intended to reassign returned value `this.removeListeners`, but forgot or something. I see that the component is getting destroyed and reassigning wouldn't do anything, but I can't be sure that you noticed the same. I can ask you questions now, but after some time you might not even remember it yourself.

My point is not to argue but to try to convince you that it's an issue that causes confusion. This one is pretty minor, but following the same pattern, one could also argue for depending on the side-effects of functions using the same "it works anyway" argument.